### ISSUE TITLE
Acceptance Testing Docs Re-organization

### ISSUE THREAD

**Issue by ThomasLandauer** [Codeception/Codeception#4394]
I'd like to re-organize http://codeception.com/docs/03-AcceptanceTests entirely.

Reasons:

* The page is too long. Even longer than http://codeception.com/docs/04-FunctionalTests or http://codeception.com/docs/05-UnitTests

* Many things are said twice. Some of them even on the very same page! And others on the AcceptanceTests page and on the WebDriver Module page.

Before I start, I'd like to have an agreement on the purpose of the page. The most important thing for me is: People need to realise that they have to make a decision (PhpBrowser or WebDriver). And for that, they need to know the (dis)advantages of both. Plus they need to see some (short) code examples, containing typical "browser" stuff, like `click`, `wait`, etc.

So my plan is:

* More comparison tables (or more infos in the present one)
* Less detail info about each of the two (PhpBrowser or WebDriver). Currently, there are ~20 code blocks for PhpBrowser! I'd say 4-5 is enough to show the idea; then refer to http://codeception.com/docs/modules/PhpBrowser for details.
* For WebDriver I'd start with ChromeDriver (standalone), cause for starters this is clearly the way to go (just 1 piece of software required!) to get a first impression of what Codeception has to offer. Then recommend Selenium for advanced users; and move all details (including installation) to WebDriver Module page.

What do you say?

**Comment by DavertMik** [comment]
Completely agree on this idea. Page should be reorganized, that's for sure. We should change priority to WebDriver.

> More comparison tables (or more infos in the present one)

I like the current one but if you know how to improve it, that would be nice

> Less detail info about each of the two (PhpBrowser or WebDriver). Currently, there are ~20 code blocks for PhpBrowser! I'd say 4-5 is enough to show the idea; then refer to http://codeception.com/docs/modules/PhpBrowser for details.

Yes!

> For WebDriver I'd start with ChromeDriver (standalone), 

I have some doubts on ChromeDriver. Yes, it is one program less to install but it is not good enough to recommend it. Selenium is very popular, as buzzword too, it is a professional tool #1 for all QA engineers, so it should be the first. Also ChromeDriver does not support smart waits.

So if you want to run real acceptance tests it is not a big deal install one tool or two.

What I'd really like to see on that page:

* reference to PageObjects in Reusing Test Code chapter
* reference to Authorization and Session Snapshots in Reusing Test Code chapter
* a complete example of an acceptance test
* AngularJS section can be removed (it's version 1.0, so it's not so important nowadays. You can just mention it once)

**Comment by ThomasLandauer** [comment]
Here's what I have so far: https://github.com/ThomasLandauer/Codeception/blob/patch-1/docs/03-AcceptanceTests.md

Questions:

1. What does `window_size: false # disabled in ChromeDriver` mean at http://codeception.com/docs/modules/WebDriver#ChromeDriver ? I just set this to `maximize` and got my window maximized! Is the info outdated (new feature in ChromeDriver maybe)?

1. For the comparison table at https://github.com/ThomasLandauer/Codeception/blob/patch-1/docs/03-AcceptanceTests.md#webdriver : Since `window_size` probably isn't true anymore, do you have some more concrete advantages of Selenium over ChromeDriver (such as SmartWait)?

1. I'd like to add a row "Available modes" to that table, with values "Headless" and "???". Is there a term for running a browser in "real" mode (i.e. not headless)?

1. Is Headless Chrome available through Selenium Server too? Or just through ChromeDriver directly?

1. Why do you favor WebDriver over PhpBrowser now? In other words: Is PhpBrowser a good option today? Or will it, with the advent of Headless Chrome, face the same fate as PhantomJS?

1. I'd like to remove everything starting from http://codeception.com/docs/03-AcceptanceTests#Click down through http://codeception.com/docs/03-AcceptanceTests#Wait-and-Act and either drop it or move it over to WebDriver/PhpBrowser Module. Is there anything that's so important that it deserves to stay here?

1. "AngularJS Testing" is going to move to WebDriver module, that's for sure. But what about "Multi Session Testing" and "Cloud Testing"?

**Comment by DavertMik** [comment]
> I just set this to maximize and got my window maximized! Is the info outdated (new feature in ChromeDriver maybe)?

What version of ChromeDriver do you use? I have 2.30 and it fails there

> Do you have some more concrete advantages of Selenium over ChromeDriver (such as SmartWait)

smartWait, popups. Also I'd add cloud based providers to this table. See *Cloud Testing* section

> I'd like to add a row "Available modes" to that table, with values "Headless" and "???". Is there a term for running a browser in "real" mode (i.e. not headless)?

window mode. I don't think this information is relevant to this. Any browser can be executed headlessly in Docker container. So it's not a problem anymore. And to my experience running Chrome in Docker is easier than setting up headless Chrome

> Is Headless Chrome available through Selenium Server too? 

Yes, it is. Both are possible: http://phptest.club/t/how-to-run-headless-chrome-in-codeception/1544

> Why do you favor WebDriver over PhpBrowser now?

Web apps use JavaScript more and more. And they need real JS testing. PhpBrowser is good for old-style interfaces, forms, etc. But for modern web development, you won't probably need it. Especially if you already covered similar cases inside functional tests.

So please concentrate on WebDriver, describe PhpBrowser in very basic manner :)

> I'd like to remove everything starting from http://codeception.com/docs/03-AcceptanceTests#Click down through http://codeception.com/docs/03-AcceptanceTests#Wait-and-Act and

No, those sections are important because from the module reference you won't easily get how to do the common stuff. For instance, in *Forms* section we show that form can be filled in and submitted in two ways: step by step actions, or using `submitForm`.

Conditional assertions are important to keep as well.

If you want to cut something, remove "Cookies, URLs, Title, etc", "Comments" sections

> "AngularJS Testing" is going to move to WebDriver module,

No, it goes to AngularJS module

> But what about "Multi Session Testing" and "Cloud Testing"

They should stay in Acceptance Testing.

**Comment by ThomasLandauer** [comment]
> What version of ChromeDriver do you use? I have 2.30 and it fails there

I have 2.30.477700 on Windows. I just tried `300x300` and `maximize`, and both are working: Chrome opens in its usual size, then resizes to what I defined.  
Maybe it's a new feature not working everywhere yet?

> popups

You're talking about popups mentioned here?: http://codeception.com/docs/modules/WebDriver#acceptPopup  So `acceptPopup` will not work with ChromeDriver?

> No, those sections are important because from the module reference you won't easily get how to do the common stuff.

Well, what's **really** important is to give an overview over all available options! Cause this is the one and only place where this is possible. So if the WebDriver Module page doesn't explain everything nicely, my solution would be to include the missing stuff there ;-)
